cartoframes/core/cartodataframe.py

```diff
@@ -4,36 +4,169 @@
 
 
 class CartoDataFrame(GeoDataFrame):
-
     def __init__(self, *args, **kwargs):
+        """
+        A CartoDataFrame object is a
+        `geopandas.GeoDataFrame <http://geopandas.org/reference/geopandas.GeoDataFrame.html>`_
+        that has a column with geometry. It extends the GeoDataFrame object to read and
+        write data from CARTO, adding wrappers when necessary.
+        """
+
         super(CartoDataFrame, self).__init__(*args, **kwargs)
 
     @staticmethod
     def from_carto(*args, **kwargs):
+        """
+        Alternate constructor to create a CartoDataFrame from a table or SQL query in CARTO.
+        Equivalent to :py:meth:`read_carto <cartoframes.io.read_carto>`.
+
+        Examples:
+
+            Using a table name:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_carto('table_name')
+
+            Using a SQL query:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_carto('SELECT * FROM table_name WHERE value > 100')
+        """
+
         from ..io.carto import read_carto
         return read_carto(*args, **kwargs)
 
     @classmethod
     def from_file(cls, filename, **kwargs):
+        """
+        Alternate constructor to create a CartoDataFrame from a file.
+        Extends from the GeoDataFrame.from_file method.
+
+        Examples:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('nybb.shp')
+        """
+
         gdf = GeoDataFrame.from_file(filename, **kwargs)
         return cls(gdf)
 
     @classmethod
     def from_features(cls, features, **kwargs):
+        """
+        Alternate constructor to create a CartoDataframe from GeoJSON features.
+        Extends from the GeoDataFrame.from_features method.
+
+        Examples:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_features('nybb.shp')
+        """
+
         gdf = GeoDataFrame.from_features(features, **kwargs)
         return cls(gdf)
 
     def to_carto(self, *args, **kwargs):
+        """
+        Upload a CartoDataFrame to CARTO.
+        Equivalent to :py:meth:`to_carto <cartoframes.io.to_carto>`.
+
+        Examples:
+
+            .. code::
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('nybb.shp')
+                cdf.to_carto(if_exists='replace')
```

Comment:
You need to specify here the name of the table: `to_carto('table', if_exists='replace')`.

NOTE: both `from_carto` and `to_carto` methods require a Credentials instance. I would add that in the docstring, and maybe the `set_default_credentials` case too.